pkg/apis/componentconfig/types.go

```diff
@@ -0,0 +1,32 @@
+/*
```

Comment:
Good question. Again, I've modeled this after k8s.

This is where k8s defines the types used to encapsulate configuration options for each component (by "component," I mean executable / process). Stuff in `cmd` populates these options from flags.

Yes... I'm blindly following how they do things.

https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/componentconfig/types.go